### PR TITLE
Filehealth cli

### PR DESCRIPTION
**Pull request by aleitner**
* Move calculations from satellite side to be done on the client side. This provides more customizability and reduces sending of unnecessary data.
* Inspector methods for checking object and segment health can both print in csv format and write to a csv file.

Segment Health
```
inspector --identity-path="satellite-identity" --address="satellite-private-address" health segment project-id -- segment-index bucket-name encrypted-path
```

Object Health
```
inspector --identity-path="satellite-identity" --address="satellite-private-address" health object project-id bucket-name encrypted-path
```

The output is by default written to `os.Stdout`, but there is also an optional `csv-path` that can be specified.

```
inspector --identity-path="satellite-identity" --address="satellite-private-address" health --csv-path="path-to-new-csv"  object project-id bucket-name encrypted-path
```
Screen Shot 2019-04-02 at 5 07 36 PM

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Could the PR be broken into smaller PRs?
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?

### CODE REVIEW

**Comment by jenlij** [comment]
Could you note the "What" and "Why" of these changes?

**Comment by aleitner** [comment]
> Could you note the "What" and "Why" of these changes?

What: Inspector tool for checking object and segment stats by a pointerdb path.
Why: Devops can use this data for tracking data stability

**Comment by jenlij** [comment]
Is there somewhere we can add the documentation above for reference since it'll be hard to find once this PR is merged?

**Comment by aleitner** [comment]
> Is there somewhere we can add the documentation above for reference since it'll be hard to find once this PR is merged?

as far as I know our code is the documentation. There's also a help command

File: cmd/inspector/main.go [hunk 1]
```diff
@@ -35,6 +36,9 @@ var (
 	// IdentityPath is the path to the identity the inspector should use for network communication
 	IdentityPath = flag.String("identity-path", "", "path to the identity certificate for use on the network")
 
+	// CSVPath is the csv path where command output is written
+	CSVPath = flag.String("csv-path", "stdout", "csv path where command output is written")
```
Comment (mniewrzal) [inline cmd/inspector/main.go]: Can you add this only to health command by `healthCmd.Flags().StringVar(&CSVPath, "csv-path", "stdout", "csv path where command output is written")` in `init` method?

Comment (egonelbre) [reply cmd/inspector/main.go]: Why do we even need an output file, people can direct the output a csv as needed.

File: cmd/inspector/main.go [hunk 2]
```diff
@@ -472,20 +476,73 @@ func ObjectHealth(cmd *cobra.Command, args []string) (err error) {
 		return ErrArgs.Wrap(err)
 	}
 
+	startAfterSegment := int64(0) // start from first segment
```
Comment (egonelbre) [inline cmd/inspector/main.go]: this file is definitely too large

Comment (aleitner) [reply cmd/inspector/main.go]: you mean inspector/main.go has too many lines?

**Pull request merged**